subsys/sal/common/sid_on_dev_cert/sid_on_dev_cert.c

```diff
@@ -633,52 +646,40 @@ sid_error_t sid_on_dev_cert_verify_and_store(void)
 		}
 	}
 
-	result = sid_pal_mfg_store_erase();
+	// init mfg_storage
+	const size_t mfg_size = tlv_flash.end_offset - tlv_flash.start_offset;
+	if (tlv_flash.storage_impl.ctx) {
+		result = flash_erase(tlv_flash.storage_impl.ctx, tlv_flash.start_offset, mfg_size);
+	}
 	if (result) {
 		LOG_ERR("MFG erase failed [%d]", result);
 		ret = SID_ERROR_STORAGE_ERASE_FAIL;
 		goto exit;
 	}
 
-	/*
-     * First of all, we need to write the MFG version.
-     * This will determine the storage format: static offsets or TLV
-     */
-	uint32_t mfg_version;
-
-	if (sid_pal_mfg_store_is_tlv_support()) {
-		mfg_version = sid_htonl(SID_PAL_MFG_STORE_TLV_VERSION);
-	} else {
-		mfg_version = sid_htonl(SID_PAL_MFG_STORE_FIXED_OFFSETS_VERSION);
-	}
-
-	result = sid_pal_mfg_store_write(SID_PAL_MFG_STORE_VERSION, (uint8_t *)&mfg_version,
-					 SID_PAL_MFG_STORE_VERSION_SIZE);
-
-	if (result) {
-		LOG_ERR("MFG version %ld is not supported", sid_htonl(mfg_version));
-		ret = SID_ERROR_NOSUPPORT;
-		goto exit;
-	}
-
+	uint8_t flags = MFG_HEADER_FLAGS_INITIALIZED;
 #ifdef CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE
-	memset(context->device_ed25519_prk, 0, SID_ODC_ED25519_PRK_SIZE);
-	memset(context->device_p256r1_prk, 0, SID_ODC_P256R1_PRK_SIZE);
-#endif /* CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE */
+	flags |= MFG_HEADER_FLAGS_KEYS_IN_PSA;
```

Comment:
One comment about flags: Can't we add a net Type for it, instaed od using the version? It will be a step towards module backward compatibility. Also the flagh might be set by '0', It can reduce flash opearion time.